base/abstractarray.jl

```diff
@@ -583,7 +583,7 @@ zero{T}(x::AbstractArray{T}) = fill!(similar(x), zero(T))
 # own, LinearSlow's CartesianRange is more complicated and requires explicit
 # inlining.
 start(A::AbstractArray) = (@_inline_meta(); itr = eachindex(A); (itr, start(itr)))
-next(A::AbstractArray,i) = (@_inline_meta(); (idx, s) = next(i[1], i[2]); (A[idx], (i[1], s)))
+next(A::AbstractArray,i) = (@_inline_meta(); (idx, s) = next(i[1], i[2]); @inbounds ret = (A[idx], (i[1], s)); ret)
```

Comment:
Noteworthy change motivated by results of performance-testing. I'm a little surprised we didn't do this before, but perhaps there some reason to think it could be problematic? CC @mbauman.